### ISSUE TITLE
[Bug]: Can't recreate any images from older versions of ControlNet from 4/29/2023, for instance, even with rollbacks.

### ISSUE THREAD

**Issue by marcsyp** [Mikubill/sd-webui-controlnet#1934]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

I have been trying to recreate images that I created on April 29th of this year using controlnet scribble_xdog, and one of my debugging activities is to try to roll ControlNet all the way back to near the date of when I created the image.  Problem is I've rolled back my ControlNet, A1111, and other repo commits to some time in April, deleted my venv folder, and reinstalled everything, and still can't recreate any of my old images.

Example.  Original Image:
![02212-2964637473-(high quality mixed media artwork_ 1 4) by (Picasso_Banksy_Jasper Johns_MC Escher_Jeff Koons_a mixed media photographer_a digita-before-highres-fix](https://github.com/Mikubill/sd-webui-controlnet/assets/2940820/22b1278d-bc9d-47e4-8df3-dd227996fac3)

Recreation (fail):
![04322-2964637473-(high quality mixed media artwork_ 1 4) by (Picasso_Banksy_Jasper Johns_MC Escher_Jeff Koons_a mixed media photographer_a digita](https://github.com/Mikubill/sd-webui-controlnet/assets/2940820/8ebba239-9a0d-44bf-a1b4-982a4c577642)

CN source f
![20230422_181025-1024 Is That Bus Chasing Me](https://github.com/Mikubill/sd-webui-controlnet/assets/2940820/5b70e5a5-65d5-442b-94c8-76cda5893535)
ile:

I have not updated any graphics drivers or changed video cards.  All on the same machine.  This example uses high-res fix but issue exists with all images.  Any ideas?

### Steps to reproduce the problem

1. Try to recreate images using parameters in metadata and source files from 4/29/2023
2. Fail miserably
3. Delete venv folder, revert to very old commits (4/15/2023 or so) in A1111 + ControlNet, disable a bunch of extensions
4. Fail miserably
5. 

### What should have happened?

well -- shouldn't I be able to recreate old images with the exact same source files and parameters??

### Commit where the problem happens

All commits

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
set PYTORCH_CUDA_ALLOC_CONF=garbage_collection_threshold:0.8,max_split_size_mb:128
set COMMANDLINE_ARGS=--opt-channelslast --use-cpu interrogate --theme dark --no-half-vae --xformers
```

### List of enabled extensions

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/2940820/26e3771f-c5ee-4157-8a44-350936ecc2f7)

### Console logs

```Shell
NONE, N/A
```

### Additional information

Example.  Original Image:
![02212-2964637473-(high quality mixed media artwork_ 1 4) by (Picasso_Banksy_Jasper Johns_MC Escher_Jeff Koons_a mixed media photographer_a digita-before-highres-fix](https://github.com/Mikubill/sd-webui-controlnet/assets/2940820/22b1278d-bc9d-47e4-8df3-dd227996fac3)

Recreation (fail):
![04322-2964637473-(high quality mixed media artwork_ 1 4) by (Picasso_Banksy_Jasper Johns_MC Escher_Jeff Koons_a mixed media photographer_a digita](https://github.com/Mikubill/sd-webui-controlnet/assets/2940820/8ebba239-9a0d-44bf-a1b4-982a4c577642)

CN source f
![20230422_181025-1024 Is That Bus Chasing Me](https://github.com/Mikubill/sd-webui-controlnet/assets/2940820/5b70e5a5-65d5-442b-94c8-76cda5893535)
ile:

**Comment by marcsyp** [comment]
Any thoughts on this?  @huchenlei maybe?  

I don't believe that I have changed any startup parameters, I'm still using the same VAEs (although VAE tests show that they are not making any significant changes), and the only thing I can thing of is that some of the underlying dependency versions may be mismatched from my original setup (boy I wish I had copied my VENV folder way back when).

Can anyone think of changes that may have resulted in drastic image consistency changes between February and April of this year?  Maybe I am misremebering the versions of torch and xformers I was on, and I need to go back even earlier??  I made these originals shortly after guess mode disappeared, I believe, but I can't be sure which versions I was on.

This is actually fairly critical to an art project that I am working on and I can't move forward until I can reproduce, sadly.

Thanks,
Marc

**Comment by marcsyp** [comment]
Maybe even some suggestions on a methodology for rolling back that I may be overlooking would be helpful.  I am willing to put in significant hours to recover this original setup.  Thanks.

**Comment by marcsyp** [comment]
One thing I've noticed is that Dynamic Prompts no longer generates the same combinations that it used to with the same seeds and parameters, and I've tried rolling that plugin version back as well to no avail.  I have not changed video cards or updated video drivers either.  Could a Windows Update have done something to change this?

**Comment by huchenlei** [comment]
Hi @marcsyp, can you share more on your setup? Maybe the metadata text on the picture?

Between Feb and April there is the ControlNet 1.1 update according to the update tracking issue https://github.com/Mikubill/sd-webui-controlnet/issues/736.

Ideally we should be able to reproduce the images, but can I ask why you want to reproduce them? Are you intending to generate images with similar art style, but different prompt(From dynamic prompt)? Will training a custom LoRA on your existing images / use reference control help maintain the style?

**Comment by marcsyp** [comment]
Thanks, yes I can share the metadata on the image (they are on the images you see above, if you want to drop them into PNG info), for instance:

![02212-2964637473-(high quality mixed media artwork_ 1 4) by (Picasso_Banksy_Jasper Johns_MC Escher_Jeff Koons_a mixed media photographer_a digita-before-highres-fix](https://github.com/Mikubill/sd-webui-controlnet/assets/2940820/ca207505-8c96-4420-8586-5361858c49d8)

(high quality mixed media artwork: 1.4) by (Picasso|Banksy|Jasper Johns|MC Escher|Jeff Koons|a mixed media photographer|a digital experience designer|an architect|a product designer|a comic book artist|a talented child: 1) in the (Street Graffiti|digital art: 1.25) style, (long shot: 0.2), (highly detailed), (masterpiece, best quality), (hyperrealistic textures: 1.25). A cat in the city at night asks the question: "Is That Bus Chasing Me?". 8k
Negative prompt: (low_quality: 1.5), (nude), pencils, deformed
Steps: 30, Sampler: Euler, CFG scale: 8, Seed: 2964637473, Size: 1024x768, Model hash: 9aba26abdf, Model: deliberate_v2, Denoising strength: 0.43, ControlNet-0 Enabled: True, ControlNet-0 Module: lineart_coarse, ControlNet-0 Model: control_v11p_sd15_lineart [43d4be0d], ControlNet-0 Weight: 0.8, ControlNet-0 Guidance Start: 0, ControlNet-0 Guidance End: 0.7, Hires upscale: 1.35, Hires steps: 12, Hires upscaler: Latent (bicubic antialiased)

This is the image before hires fix, as above.  In this case, I'm not and was not using dynamic prompts, so I don't believe that should be a factor here.

I need to recreate the exact image (or very close, some pixel variance OK, but major content variance not) for a variety of purposes:

1) Animations using CN Weight travel (using XYZ plot creatively), prompt travel, controlnet travel, seed travel
2) Recreating very similar images with the highest fidelity using the same model and parameters but using a different control image, and this one is a client request.  I have used the image with Reference processors and I can get a "sort of" result, but it is far from what I am looking for.  

Notes:
I don't believe a Lora will work too well because in most cases, I am heavily using dynamic prompting and the style of my images is very very different, and I am looking to recreate the look and feel of one specific image.
I rolled back to before CN 1.1, but that didn't seem to affect anything.  However, my method of rollback is to delete the VENV folder, checkout a bunch of commits from roughly the time period I am hoping to recreate, and then to launch using webui-user and let it reinstall everything.  However, I am concerned that there is a requirements file somewhere that is not being reverted properly and I am getting newer versions of dependencies all over the place, and this may be the root of the problem.  That's just a gut feeling though.

Thoughts?
